devops/linux/docker/deploy-local.sh

```diff
@@ -0,0 +1,162 @@
+JEMPI_CONFIGURATION_PATH=$JEMPI_HOME/JeMPI_Apps/JeMPI_Configuration/reference/config-reference.json
+
+# Display menu options
+echo "Select an option for local deployment:"
+echo "1. Deploy JeMPI from Scratch (With all installations...)."
+echo "2. Build and Reboot."
+echo "3. Restart JeMPI."
+echo "4. Down the JeMPI."
+echo "5. Destroy JeMPI (This process will wipe all data)."
+
+
+
+
+# Prompt user for choice
+read -p "Enter your choice (1-5): " choice
+
+
+install_docker() {
+    if command -v docker &> /dev/null; then
+        echo "Docker is already installed."
+    else
+        # Install Docker
+        echo "Installing Docker... "
+        sudo apt-get update
+        sudo apt-get install -y docker.io
+
+        # Add your user to the docker group to run Docker without sudo
+        sudo usermod -aG docker $USER
+
+        echo "Docker has been installed."
+    fi
+}
+
+# Function to install SDKMAN, Java, Maven, and SBT
+install_sdkman_and_java_sbt_maven() {
+    # Install SDKMAN
+    echo "Installing SDKMAN... "
+    curl -s "https://get.sdkman.io" | bash
+
+    # Initialize SDKMAN
+    source "$HOME/.sdkman/bin/sdkman-init.sh"
+
+    # Install Java, Maven, and SBT using SDKMAN
+    echo "Installing Java : $JAVA_VERSION ... "
+    sdk install java $JAVA_VERSION
+    sdk use java $JAVA_VERSION
+
+    echo "Installing maven... "
+    sdk install maven
+
+    echo "Installing sbt... "
+    sdk install sbt
+}
+run_enviroment_configuration_and_helper_script(){
+    # Navigate to environment configuration directory
+    echo "Navigate to environment configuration directory"
+    cd $JEMPI_HOME/devops/linux/docker/conf/env/
+    source $JEMPI_HOME/devops/linux/docker/conf/env/create-env-linux-low-1.sh
+
+    # Running Docker helper scripts 
+    echo "Running Docker helper scripts "
+    cd $JEMPI_HOME/devops/linux/docker/helper/scripts/
+    source $JEMPI_HOME/devops/linux/docker/helper/scripts/x-swarm-a-set-insecure-registries.sh
+}
+
+run_field_configuration_file() {
+    # Running Docker helper scripts
+    echo "Running JeMPI configuration with path: $JEMPI_CONFIGURATION_PATH"
+    cd $JEMPI_HOME/JeMPI_Apps/JeMPI_Configuration/
+    source $JEMPI_HOME/JeMPI_Apps/JeMPI_Configuration/create.sh $JEMPI_CONFIGURATION_PATH   
+}
+
+initialize_swarm(){
+    if docker info | grep -q "Swarm: active"; then
+        echo "Docker Swarm is running."
+    else
+        echo "Docker Swarm is not running."
+        echo "Initialize Swarm on node1"
+         cd $JEMPI_HOME/devops/linux/docker
+        source $JEMPI_HOME/devops/linux/docker/b-swarm-1-init-node1.sh
+
+    fi
+}
+
+pull_docker_images_and_push_local(){
+    # Navigate to Docker directory
+    cd $JEMPI_HOME/devops/linux/docker
+    echo "Create Docker registry"
+    source $JEMPI_HOME/devops/linux/docker/c-registry-1-create.sh
+
+    # Create Docker registry
+    if docker images registry | grep -q 'registry'; then
```

Comment:
Should you not source the `devops/linux/docker/c-registry-1-create.sh` (above) after checking this. I think as it is, it will always be true.

Also you could **not** the if statement and add the logic there, as opposed to the empty if statement